worker/src/main/scala/org/apache/celeborn/service/deploy/worker/PushDataHandler.scala

```diff
@@ -1068,6 +1074,6 @@ class PushDataHandler extends BaseMessageHandler with Logging {
 }
 
 object PushDataHandler {
-  @volatile var pushMasterDataTimeoutTested = false
-  @volatile var pushSlaveDataTimeoutTested = false
+  @volatile private var pushMasterMergeDataTimeoutTested = false
```

Comment:
I would like to inquire about the reasoning behind choosing the `pushMasterMergeDataTimeoutTested` as an object variable instead of a class variable. @waitinfuture

Reply:
If I were to move `pushMasterMergeDataTimeoutTested ` to a class variable, it would result in a failure of the CI.

Reply:
I dived into it and think the reason is that, if put the variables in the object, it's process-level singleton, as a result, although we have 3 workers in the MiniCluster, only one timeout will be triggered, and the worker will be put in the blacklist. However, when put the variables into Handler, all three workers will trigger timeout and all will be blacklist, as a result it will cause Revive Exception for no slots available. Following are related logs:
```
Reporting Worker Failure:
Host:30.221.101.208:RpcPort:57565:PushPort:57568:FetchPort:57570:ReplicatePort:57569   PUSH_DATA_TIMEOUT_MASTER   1687321430258
Current blacklist:
Host:30.221.101.208:RpcPort:57560:PushPort:57562:FetchPort:57564:ReplicatePort:57563   PUSH_DATA_TIMEOUT_MASTER   1687321430254
Host:30.221.101.208:RpcPort:57572:PushPort:57576:FetchPort:57578:ReplicatePort:57577   PUSH_DATA_TIMEOUT_MASTER   1687321430248  
```
```
Revive failed, status SLOT_NOT_AVAILABLE
```

Reply:
thanks for the detailed explanation.

Reply:
then the variable `pushMasterDataTimeoutTested` and `pushSlaveDataTimeoutTested` should also be a singleton.

Reply:
The reason behind the successful passing of the `PushDataTimeoutTest` is that the `PushDataHandler#handlePushData` is not triggered. This is due to the fact that the current test case consistently ensures that the buffer size of the hash-based shuffle writer remains below the `PUSH_BUFFER_MAX_SIZE`.

Reply:
Oh, that's unexpected, thanks for your exploration. So I think we should make it consistent for ```testPushMasterDataTimeout/testPushSlaveDataTimeout``` and ```pushMasterMergeDataTimeoutTested/pushSlaveMergeDataTimeoutTested```

Reply:
ok, will raise a follow-up